src/classes/RD2_DataMigrationMapper.cls

```diff
@@ -159,6 +161,20 @@ public class RD2_DataMigrationMapper {
         }
     }
 
+    /**
+     * @description Determines whether Amount can be calculated.  Throws an exception if not.
+     * @param rd Recurring Donation
+     */
+    private void validateAmountCanBeCalculated(npe03__Recurring_Donation__c rd) {
+
+        if (rd.npe03__Open_Ended_Status__c == RD_Constants.OPEN_ENDED_STATUS_CLOSED
```

Comment:
Based on the `setOtherFields()` method, a fixed RD is the one with `rd.npe03__Open_Ended_Status__c == RD_Constants.OPEN_ENDED_STATUS_NONE || rd.npe03__Open_Ended_Status__c == null` so when an RD is closed, we do not know if it was a fixed or an open RD. Based on the `setOtherFields()`, the "Number of Planned Installments"(`npe03__Installments__c`) can be populated or null for an open RD. Thus, I think we also need to check that `rd.npe03__Installments__c != null`. The lack of that check is the reason why the existing test is failing.

Reply:
I'm not sure I agree with the conclusion here.  I think the failing test no longer represents a valid outcome.  Let's discuss offline during dev office hours.

Reply:
A closed RD can be a fixed or open RD before being closed. If an open legacy RD is closed and "Number of Planned Installments" is not provided (null), then this new validation check will fail validation/migration process but it should not be.

Reply:
My reading of the failing test is that it precisely creates the conditions for AC1 of the current work item.  And that AC specifies that the migration should fail.

Reply:
I've concluded that my understanding of the requirements was incorrect and your interpretation is correct.  I've updated the code to reflect your original suggestion.